text/0000-payment-proofs.md

```diff
@@ -0,0 +1,101 @@
+- Title: payment-proofs
+- Authors: [David Burkett](mailto:davidburkett38@gmail.com)
+
+---
+
+# Summary
+[summary]: #summary
+
+Support generating and validating payment proofs for sender-initiated (ie. non-invoice) transactions. 
+
+# Motivation
+[motivation]: #motivation
+
+Bitcoin and other cryptocurrencies with transparent protocol-level addressing and immutable, unpruneable blockchains can prove sender, receiver, and amounts of payments simply by pointing to the transaction in the blockchain.
+Grin's privacy and scalability means users no longer have this ability. This prevents some merchants from accepting Grin due to the high possibility of payment disputes that are unresolvable in the same way they are for transparent coins.
+
+This RFC proposes a change to the transaction building process where payers can require payees to create a "proof" they've received a payment before the payer finalizes and broadcasts the transaction.
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+From an end-user perspective, payers can require payees to prove receipt of funds as part of the transacting process.
+Payers can then use these "proofs", along with information from the blockchain, to resolve payment disputes and prove funds were sent to the correct payee, and the transaction was confirmed on the blockchain.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Slate changes
+
+A new (optional) structure (`payment_info`) will be added to transaction slates, along with a version increase. The `payment_info` structure will contain:
+
+* `message` - Optional message to include in the payment proof. Limited to 100(?) characters.
```

Comment:
IIRC there already is a message field in the slate. Is this a second message field? Is it possible to use the existing one?

Reply:
This is a separate one. Unable to reuse the existing, because the existing requires a signature from an address on a different elliptic curve. We can possibly deprecate the existing one, because tbh, I'm not sure there's an actual use-case for it.

Reply:
I think we should deprecate the existing message field in favor of this one

Reply:
Coming back to this conversation based on what @CircusDad wrote below, what is the motivation for this message field, and is it possible the existing message field could be good enough?

I think there's a valid point in what is raised in that we wouldn't want to inadvertently create binding agreements in slates, doesn't seem to be the right place for this.